src/services/requestSessions.js

```diff
@@ -0,0 +1,710 @@
+				jobId,
+				delay,
+				'ExpireSessionRequest',
+				reqBody,
+				`${common.expireSessionRequestEndpoint}/${SessionRequestResult.id}`,
+				common.POST_METHOD
+			)
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.created,
+				message: 'SESSION_REQUEST_SENT_SUCCESSFULLY',
+				result: SessionRequestResult,
+			})
+		} catch (error) {
+			console.error(error)
+			throw error
+		}
+	}
+
+	/**
+	 * Get a list of pending session requests for a user.
+	 * @param {string} userId - The ID of the user.
+	 * @param {number} pageNo - The page number for pagination.
+	 * @param {number} pageSize - The number of records per page.
+	 * @returns {Promise<Object>} The list of pending session requests.
+	 */
+	static async list(userId, pageNo, pageSize, status) {
+		try {
+			const allRequestSession = await sessionRequestQueries.getAllRequests(userId, status)
+
+			const sessionRequestData = allRequestSession.rows.map((session) => session)
+
+			const sessionRequestMapping = await sessionRequestMappingQueries.getSessionsMapping(userId)
+
+			const sessionRequestIds = sessionRequestMapping.map((session) => session.session_request_id)
+
+			const sessionMappingDetails = await sessionRequestQueries.getSessionMappingDetails(
+				sessionRequestIds,
+				status
+			)
+
+			const sessionMappingDetailsData = sessionMappingDetails.map((session) => session.dataValues)
+
+			const combinedData = [...sessionRequestData, ...sessionMappingDetailsData]
+
+			const totalCount = combinedData.length
+
+			// Apply pagination
+			const offset = (pageNo - 1) * pageSize
+			const paginatedData = combinedData.slice(offset, offset + pageSize)
+
+			if (paginatedData.length === 0) {
+				return responses.successResponse({
+					statusCode: httpStatusCode.ok,
+					message: 'SESSION_REQUESTS_LIST',
+					result: { data: [], count: 0, pageNo, pageSize },
+				})
+			}
+
+			// Get opposite user ID for each session
+			const oppositeUserIds = paginatedData.map((session) =>
+				session.requestor_id === userId ? session.requestee_id : session.requestor_id
+			)
+
+			// Fetch user details for opposite users
+			let oppositeUserDetails = await userExtensionQueries.getUsersByUserIds(oppositeUserIds, {
+				attributes: ['user_id', 'image', 'name', 'experience', 'designation', 'organization_id'],
+			})
+
+			const userExtensionsModelName = await userExtensionQueries.getModelName()
+			const uniqueOrgIds = [...new Set(oppositeUserDetails.map((obj) => obj.organization_id))]
+
+			oppositeUserDetails = await entityTypeService.processEntityTypesToAddValueLabels(
+				oppositeUserDetails,
+				uniqueOrgIds,
+				userExtensionsModelName,
+				'organization_id'
+			)
+
+			const userDetailsMap = oppositeUserDetails.reduce((acc, user) => {
+				acc[user.user_id] = user
+				return acc
+			}, {})
+
+			const userIds = oppositeUserIds.map((id) => String(id))
+			const userDetails = await userRequests.getListOfUserDetails(userIds, true)
```

Comment:
@sumanvpacewisdom we should't call user service, get the data from mentoring db itself